src/Neo/Hardfork.cs

```diff
@@ -12,7 +12,8 @@ namespace Neo
 {
     public enum Hardfork : byte
     {
-        HF_Aspidochelone,
-        HF_Basilisk
+        HF_Genesis = 0,
+        HF_Aspidochelone = 1,
```

Comment:
Wow, that changes it to 1 from 0, I'm not sure what can be affected by this.

Reply:
i dont think it is a good idea to change the public enum value, private one is OK, public one should be avoided since it might be used as a `byte` by others

Reply:
https://github.com/neo-project/neo/blob/4788421cbb0a56a051ac9e5e101e4514d50f4150/src/Neo/SmartContract/Native/NativeContract.cs#L134

I think that we only use strings for rpc and json, and maybe we can use this value for the method version

Reply:
I think that it was a bad idea to use 0 for different than `None` or `Genesis` from the beginning, I think that we never use `HardForks` integer value, just for ordering.

Reply:
> we never use HardForks integer value, just for ordering.

Agree, this change is expected to be safe, NeoGo (and existing configurations) also are not affected by this, we parse them from strings. But I support the solution with nullable `ActiveIn` to avoid hardforks enum modification.

Reply:
> It is not about whether it is safe for us to do so, this is not only about few of us who is maintaining the core, but the whole community who are also developing their tools. If we just change existing public data types at well, things will be problematic for others to follow.

@Liaojinghui, if your comment belongs to this conversation, then it's a translation inaccuracy from my side and here by "us" I meant both developers and node operators/users. E.g. NeoGo operators or users do not use `byte` representation of hardfork enum; all user-facing hardforks have string representation (both in the node config and RPC). Thus, users won't be affected by this change. And as Fernando said, the same thing is likely relevant for NeoC# node.